homeassistant/helpers/entity_registry.py

```diff
@@ -147,18 +147,25 @@ def async_get_or_create(
                 ),
             )
 
+        entry = self.hass.config_entries.async_get_entry(config_entry_id)
```

Comment:
I think that we should just change it that we pass the whole entry into this method instead of just the config entry id? that saves a lookup.